extensions/interactions/rules.json

```diff
@@ -0,0 +1,186 @@
+{
+  "CodeRepl": {
```

Comment:
Yeah, my response was hinging on details I didn't quite understand. :) Having a separate rules.json file per interaction seems just as reasonable, so yep I like it.